crates/bevy_reflect/bevy_reflect_derive/src/utility.rs

```diff
@@ -21,3 +21,78 @@ pub(crate) fn get_reflect_ident(name: &str) -> Ident {
     let reflected = format!("Reflect{}", name);
     Ident::new(&reflected, Span::call_site())
 }
+
+/// Helper struct used to process an iterator of `Result<Vec<T>, syn::Error>`,
+/// combining errors into one along the way.
+pub(crate) struct ResultSifter<T> {
+    items: Vec<T>,
+    errors: Option<syn::Error>,
+}
+
+/// Returns a `Member` made of `ident` or `index` if `ident` is None.
+///
+/// Rust struct syntax allows for `Struct { foo: "string" }` with explicitly
+/// named fields. It allows the `Struct { 0: "string" }` syntax when the struct
+/// is declared as a tuple struct.
+///
+/// ```
+/// # fn main() {
+/// struct Foo { field: &'static str }
+/// struct Bar(&'static str);
+/// let Foo { field } = Foo { field: "hi" };
+/// let Bar { 0: field } = Bar { 0: "hello" };
+/// let Bar(field) = Bar("hello"); // more common syntax
+/// # }
+/// ```
+///
+/// This function helps field access in context where you are declaring either
+/// a tuple struct or a struct with named fields. If you don't have a field name,
+/// it means you need to access the struct through an index.
+pub(crate) fn ident_or_index(ident: Option<&Ident>, index: usize) -> Member {
+    // TODO(Quality) when #4761 is merged, code that does this should be replaced
```

Comment:
Is this TODO comment still relevant? This seems self-referential.

Reply:
Yeah there is one other place that could use this utility function, but we chose to do that later so that we didn't increase the diff of this already-massive PR.